### PR TITLE
feat:  Add initial SQS queues

### PR DESCRIPTION
**Pull request by olevett**
further work on #34

### CODE REVIEW

**Comment by olevett** [comment]
We don't directly actually need the DLQ for the Audit flow, and as per #46 , I think there's a neater long term way of doing this.

File: terraform/modules/data-share-service/sqs.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module "data_processor_queue" {
+  source      = "../sqs"
+  queue_name  = "${var.environment}-gdx-data-share-poc-data-processor"
```
Comment (oskarwilliams378) [inline terraform/modules/data-share-service/sqs.tf:3]: the others all end in queue?

File: terraform/modules/sqs/main.tf [hunk 2]
```diff
@@ -0,0 +1,69 @@
+data "aws_caller_identity" "current" {}
+
+# rework as part of #46
+#tfsec:ignore:aws-iam-no-user-attached-policies 
+resource "aws_iam_user" "sqs_user" {
+  name = "${var.environment}-sqs-queue-${var.queue_name}"
```
Comment (oskarwilliams378) [inline terraform/modules/sqs/main.tf:6]: is this going to have the env in twice then? maybe just do 
```suggestion
  name = "${var.queue_name}-sqs-queue-user"
```

File: terraform/modules/sqs/main.tf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  }
+}
+
+resource "aws_iam_user_policy" "sqs_policy" {
+  name   = "{var.environment}-sqs-queue-${var.queue_name}"
+  policy = data.aws_iam_policy_document.sqs_policy.json
+  user   = aws_iam_user.sqs_user.name
+}
+
+resource "aws_kms_key" "kms_key" {
```
Comment (oskarwilliams378) [inline terraform/modules/sqs/main.tf:34]: aliases maybe? (see #48 )

**Review by oskarwilliams378 — APPROVED** [review]
couple v minor things, not blocking

**Pull request merged**